### PR TITLE
[Transform] Allow specifying destination index aliases in the Transform's dest config

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR introduces the possibility of specifying aliases for the destination index.
The aliases are specified as part of `TransformConfig` on `PUT`:
```
{
  ...
  "dest": {
    "index": "my-transform-dest",
    "aliases": [
      {
        "alias": "my-dest-alias-1"
      },
      {
        "alias": "my-dest-alias-2",
        "move_on_creation": true
      }
    ]
  },
  ...
```

When the aliases are specified, the destination index is added to all the aliases, regardless whether the destination index has just been created by the transform or pre-created by the user.
Additionally, for the aliases with `move_on_creation` set to `true`, all the other indices are removed from the alias, ensuring that the alias points at only one index (i.e.: the destination index of the current transform).

Relates https://github.com/elastic/elasticsearch/issues/93796

### CODE REVIEW

File: docs/changelog/94943.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 94943
+summary: Allow specifying destination index intended aliases in the Transform's dest
+  config
```
Comment (droberts195) [inline docs/changelog/94943.yaml:3]:
```suggestion
summary: Allow specifying destination index aliases in the Transform's `dest` config
```

**Review by droberts195 — COMMENTED** [review]
The docs for the create transform API should include this new option. A key piece of information to include in these docs is that the aliases are manipulated using the stored credentials of the transform, which means the secondary credentials supplied at creation time (if both primary and secondary credentials are specified).

File: x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransformPrivilegeCheckerTests.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public class TransformPrivilegeCheckerTests extends ESTestCase {
         .build();
     private static final TransformConfig TRANSFORM_CONFIG = new TransformConfig.Builder().setId(TRANSFORM_ID)
         .setSource(new SourceConfig(SOURCE_INDEX_NAME))
-        .setDest(new DestConfig(DEST_INDEX_NAME, null))
+        .setDest(new DestConfig(DEST_INDEX_NAME, null, null))
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/action/TransformPrivilegeCheckerTests.java:83]: > Do we need to be testing for extra privileges in the privileges checker itself if aliases are specified?

Yes, it turns out that we need to.
As per documentation, in order to apply alias actions the user needs to have `manage` privilege on both index **and** the alias.
I've just added this check to `TransformPrivilegeChecker`.
Additionally, I added:
+ unit test in `TransformPrivilegeCheckerTests`
+ integration test in `TransformInsufficientPermissionsIT` for the case where this privilege is missing.

> If so we should only require these privileges if aliases are specified.

Ack. That's what the `TransformPrivilegeChecker` does now.

**Comment by przemekwitek** [comment]
> aliases are manipulated using the stored credentials of the transform, which means the secondary credentials supplied at creation time (if both primary and secondary credentials are specified).

Done.

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**